java-frontend/src/main/java/org/sonar/java/se/xproc/BehaviorCache.java

```diff
@@ -193,4 +218,22 @@ private static MethodBehavior createGuavaPreconditionsBehavior(Symbol.MethodSymb
     behavior.completed();
     return behavior;
   }
+
+  private static MethodBehavior createCollectionUtilsBehavior(Symbol.MethodSymbol symbol) {
+    MethodBehavior behavior = new MethodBehavior(symbol);
+
+    HappyPathYield happyPathYield = new HappyPathYield(behavior);
+    happyPathYield.parametersConstraints.add(pmapForConstraint(ObjectConstraint.NULL));
+    happyPathYield.setResult(-1, pmapForConstraint(BooleanConstraint.TRUE));
+    behavior.addYield(happyPathYield);
+
+    happyPathYield = new HappyPathYield(behavior);
+    happyPathYield.parametersConstraints.add(pmapForConstraint(ObjectConstraint.NOT_NULL));
+    happyPathYield.setResult(-1, PCollections.emptyMap());
```

Comment:
Why not 2 explicit happy path yields for `NOT_NULL` and results being `TRUE` and `FALSE` ?